plotting/plot_factory/plot_factory.py

```diff
@@ -0,0 +1,208 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Constructs a plot and DashApp object for insertion into directly into a web page
+"""
+
+# Note pandas and plotly.graph_objs dependencies are used, but not recognised by pycharm interpreter
+
+# Visualisation Dependencies
+import dash
+import dash_core_components as dcc
+import dash_html_components as html
+import plotly.graph_objects as go  # pylint: disable=unused-import
+
+# Internal Dependencies
+from plotting.plot_meta_language.interpreter import Interpreter
+
+
+# Data Dependencies
+
+
+class PlotFactory:
+    """Creates figures from formatted data and layout
+       producing a DashApp for direct insertion inside a given web page
+       =_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=_=
+        """
+
+    @staticmethod
+    def get_trace_list(data, layout, figure_name):
+        """Creates trace list containing traces for each spectrum to place in figure
+
+        :param data
```

Comment:
Need to add explanations of what the parameters are e.g.:
:param figure_name: (str) The name of the figure